ui/Model.elm

```diff
@@ -126,42 +159,100 @@ getReleaseList =
         |> Kinto.send ReleasesFetched
 
 
-extractChannels : List Release -> Dict.Dict Channel Bool
-extractChannels releaseList =
-    List.map (.details >> .channel >> (\c -> ( c, True ))) releaseList
+extractFilterSet : (Release -> String) -> List Release -> FilterSet
+extractFilterSet accessor releases =
+    List.map (accessor >> (\c -> ( c, True ))) releases
         |> Dict.fromList
 
 
-filterReleases : Model -> List Release
-filterReleases { filters, releases } =
+extractVersionFilterSet : List Release -> FilterSet
+extractVersionFilterSet releases =
+    -- TODO: sort desc
+    let
+        extractVersion ( version, x ) =
+            case (String.split "." version) of
+                v :: _ ->
+                    ( v, x )
+
+                _ ->
+                    ( "None", x )
+    in
+        extractFilterSet (.details >> .filename) releases
```

Comment:
Fixed in https://github.com/leplatrem/systemaddons-versions/pull/2/commits/832a3bdede94cf68dc4975b60d4debcc0c0aebfa